src/IcedTasks/CancellableTask.fs

```diff
@@ -545,56 +545,18 @@ module CancellableTasks =
             /// </summary>
             [<NoEagerConstraintApplication>]
-            static member inline BindDynamic< ^TaskLike, 'TResult1, 'TResult2, ^Awaiter, 'TOverall
-                when ^TaskLike: (member GetAwaiter: unit -> ^Awaiter)
-                and ^Awaiter :> ICriticalNotifyCompletion
-                and ^Awaiter: (member get_IsCompleted: unit -> bool)
-                and ^Awaiter: (member GetResult: unit -> 'TResult1)>
-                (
-                    sm: byref<ResumableStateMachine<CancellableTaskStateMachineData<'TOverall>>>,
-                    [<InlineIfLambda>] task: CancellationToken -> ^TaskLike,
-                    continuation: ('TResult1 -> CancellableTaskCode<'TOverall, 'TResult2>)
-                ) : bool =
-                sm.Data.ThrowIfCancellationRequested()
-
-                let mutable awaiter =
-                    (^TaskLike: (member GetAwaiter: unit -> ^Awaiter) (task
-                                                                           sm.Data.CancellationToken))
-
-                let cont =
-                    (CancellableTaskResumptionFunc<'TOverall>(fun sm ->
-                        let result = (^Awaiter: (member GetResult: unit -> 'TResult1) (awaiter))
-                        (continuation result).Invoke(&sm)
-                    ))
-
-                // shortcut to continue immediately
-                if (^Awaiter: (member get_IsCompleted: unit -> bool) (awaiter)) then
-                    cont.Invoke(&sm)
-                else
-                    sm.ResumptionDynamicInfo.ResumptionData <-
-                        (awaiter :> ICriticalNotifyCompletion)
-
-                    sm.ResumptionDynamicInfo.ResumptionFunc <- cont
-                    false
-
-
-            /// <summary>
-            /// The entry point for the dynamic implementation of the corresponding operation. Do not use directly, only used when executing quotations that involve tasks or other reflective execution of F# code.
-            /// </summary>
-            [<NoEagerConstraintApplication>]
-            static member inline BindDynamic< ^TaskLike, 'TResult1, 'TResult2, ^Awaiter, 'TOverall
-                when ^TaskLike: (member GetAwaiter: unit -> ^Awaiter)
-                and ^Awaiter :> ICriticalNotifyCompletion
+            static member inline BindDynamic<'TResult1, 'TResult2, ^Awaiter, 'TOverall
+                when ^Awaiter :> ICriticalNotifyCompletion
                 and ^Awaiter: (member get_IsCompleted: unit -> bool)
                 and ^Awaiter: (member GetResult: unit -> 'TResult1)>
                 (
                     sm: byref<ResumableStateMachine<CancellableTaskStateMachineData<'TOverall>>>,
-                    task: ^TaskLike,
+                    [<InlineIfLambda>] getAwaiter: CancellationToken -> ^Awaiter,
```

Comment:
Not sure this makes a difference, since the argument is already static (i.e., it uses an SRTP constraint, so it has to be inlined). Did you see a difference in the generated code?

Reply:
No I still haven't gotten around to investigating the different state machine generation and yeah I'm unsure how often I'd actually need `InlineIfLambda` if at all in most cases in this file anyway.

Reply:
For (some) insights, see this answer by Don: https://github.com/dotnet/fsharp/discussions/14123#discussioncomment-3983233